server/subscribe.go

```diff
@@ -360,22 +365,95 @@ func (p *Plugin) listChannelSubscriptions() (string, error) {
 		return "", err
 	}
 
+	teamDMGMSubs, err := p.getSubscriptionsOrderedByTeamDMandGM()
+	if err != nil {
+		return "", err
+	}
+
 	rows := []string{}
+	printDMGMHeader := true
+
+	for _, teamSubs := range teamDMGMSubs {
+
+		// create team header for channels.  Only print header for DMs and GMs channels once.
+		if teamSubs.TeamId != "" {
+			channelTeam, appErr := p.API.GetTeam(teamSubs.TeamId)
+			if appErr != nil {
+				return "", appErr
+			}
+			rows = append(rows, fmt.Sprintf("\n### %s", channelTeam.DisplayName))
+		} else {
+			if printDMGMHeader {
```

Comment:
Why is this boolean necessary? Do we traverse into this `else` more than once throughout the loop?

Reply:
This has been completely removed from the function.  The new methodology has the `GetSubscriptions()` method do all of the work so that `listChannelSubscriptions` just loops through the data.